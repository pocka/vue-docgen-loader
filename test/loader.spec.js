import compiler from './compiler'

it('Injects docgen result as __docgenInfo property', async () => {
  const stats = await compiler('./fixtures/basic-sfc.vue')
  const output = stats.toJson().modules[0].modules[0].source

  const docgenPattern = /\.__docgenInfo\s?=\s?(\{[\s\S]*})/

  expect(output).toMatch(docgenPattern)
  expect(JSON.parse(output.match(docgenPattern)[1])).toMatchSnapshot()
})

it('Injects docgen result for non-SFC', async () => {
  const fixture = './fixtures/basic.vue.js'

  const stats = await compiler(fixture)
  const output = stats.toJson().modules.find(mod => mod.name.includes(fixture))
    .source

  const docgenPattern = /\.__docgenInfo\s?=\s?(\{[\s\S]*})/

  expect(output).toMatch(docgenPattern)
  expect(JSON.parse(output.match(docgenPattern)[1])).toMatchSnapshot()
})

it('Injects docgen result non-SFC with multiple exports', async () => {
  const fixture = './fixtures/basicMulti.vue.js'

  const stats = await compiler(fixture)
  const output = stats.toJson().modules.find(mod => mod.name.includes(fixture))
    .source

  const buttonExports = ['MyButton1', 'MyButton2']

  const createDocgenPattern = exportName =>
    `^.*${exportName}.*\\.__docgenInfo\\s?=\\s?(\\{[\\s\\S]*?})$`
  const match1 = new RegExp(createDocgenPattern(buttonExports[0]), 'm')
  const match2 = new RegExp(createDocgenPattern(buttonExports[1]), 'm')

  expect(output).toMatch(match1)
  expect(output).toMatch(match2)
  expect(JSON.parse(output.match(match1)[1])).toMatchSnapshot()
  expect(JSON.parse(output.match(match2)[1])).toMatchSnapshot()
})
