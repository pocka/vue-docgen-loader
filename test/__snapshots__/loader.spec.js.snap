// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Injects docgen result as __docgenInfo property 1`] = `
Object {
  "description": "",
  "displayName": "basic-sfc",
  "exportName": "default",
  "props": Array [
    Object {
      "defaultValue": Object {
        "func": false,
        "value": "'bar'",
      },
      "description": "Description of foo.",
      "name": "foo",
      "tags": Object {},
      "type": Object {
        "name": "string",
      },
    },
  ],
  "tags": Object {},
}
`;

exports[`Injects docgen result for non-SFC 1`] = `
Object {
  "description": "",
  "displayName": "basic.vue",
  "exportName": "MyButton",
  "props": Array [
    Object {
      "defaultValue": Object {
        "func": false,
        "value": "-1",
      },
      "description": "Foo the number.",
      "name": "foo",
      "tags": Object {},
      "type": Object {
        "name": "number",
      },
    },
  ],
  "tags": Object {},
}
`;

exports[`Injects docgen result non-SFC with multiple exports 1`] = `
Object {
  "description": "",
  "exportName": "MyButton1",
  "props": Array [
    Object {
      "defaultValue": Object {
        "func": false,
        "value": "-1",
      },
      "description": "Foo the number.",
      "name": "foo",
      "tags": Object {},
      "type": Object {
        "name": "number",
      },
    },
  ],
  "tags": Object {},
}
`;

exports[`Injects docgen result non-SFC with multiple exports 2`] = `
Object {
  "description": "",
  "exportName": "MyButton2",
  "props": Array [
    Object {
      "defaultValue": Object {
        "func": false,
        "value": "-1",
      },
      "description": "Foo the number.",
      "name": "foo",
      "tags": Object {},
      "type": Object {
        "name": "number",
      },
    },
  ],
  "tags": Object {},
}
`;
